Description: Disable runpath checking
Author: Debian Cyrus Team <pkg-cyrus-imapd-debian-devel@lists.alioth.debian.org>
Forwarded: no
Reviewed-By: Xavier Guimard <yadd@debian.org
Last-Update: 2020-02-10

---
 configure.ac | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 5319b2512..8030a6575 100644
--- a/configure.ac
+++ b/configure.ac
@@ -189,9 +189,6 @@ fi
 dnl Check the required alignment for various types
 AC_CHECK_ALIGNOF(uint32_t)
 
-dnl check for -R, etc. switch
-CMU_GUESS_RUNPATH_SWITCH
-
 AC_CHECK_HEADERS(unistd.h sys/select.h sys/param.h stdarg.h)
 AC_REPLACE_FUNCS(memmove strcasecmp ftruncate strerror posix_fadvise strsep memmem memrchr)
 AC_CHECK_FUNCS(strlcat strlcpy strnchr getgrouplist fmemopen pselect futimens futimes)
