Description: Fix segfault on mailshare sive redirect
Author: Anthony Prades <https://github.com/toony>
Origin: upstream, https://github.com/cyrusimap/cyrus-imapd/commit/7a11575cadca55630a37f8c07298aae4b4cf2a4a
Bug: https://github.com/cyrusimap/cyrus-imapd/pull/2597
Bug-Debian: https://bugs.debian.org/927142
Forwarded: not-needed
Reviewed-By: Xavier Guimard <yadd@debian.org>
Last-Update: 2019-05-09

--- a/imap/lmtp_sieve.c
+++ b/imap/lmtp_sieve.c
@@ -414,7 +414,7 @@
     /* if we have a msgid, we can track our redirects */
     if (m->id) {
         snprintf(buf, sizeof(buf), "%s-%s", m->id, rc->addr);
-        sievedb = make_sieve_db(mbname_userid(sd->mbname));
+        sievedb = make_sieve_db(mbname_recipient(sd->mbname, ((deliver_data_t *) mc)->ns));
 
         dkey.id = buf;
         dkey.to = sievedb;
@@ -496,7 +496,7 @@
     body = msg_getheader(md, "original-recipient");
     origreceip = body ? body[0] : NULL;
     if ((res = send_rejection(md->id, md->return_path,
-                              origreceip, mbname_userid(sd->mbname),
+                              origreceip, mbname_recipient(sd->mbname, ((deliver_data_t *) mc)->ns),
                               rc->msg, md->data)) == 0) {
         snmp_increment(SIEVE_REJECT, 1);
         syslog(LOG_INFO, "sieve rejected: %s to: %s",
@@ -735,7 +735,7 @@
     while (waitpid(sm_pid, &sm_stat, 0) < 0);
 
     if (sm_stat == 0) { /* sendmail exit value */
-        sievedb = make_sieve_db(mbname_userid(sdata->mbname));
+        sievedb = make_sieve_db(mbname_recipient(sdata->mbname, ((deliver_data_t *) mc)->ns));
 
         dkey.id = outmsgid;
         dkey.to = sievedb;
