--- a/imap/quota_db.c
+++ b/imap/quota_db.c
@@ -89,7 +89,15 @@ int quota_read(struct quota *quota, stru
 
     switch (r) {
     case CYRUSDB_OK:
-	sscanf(data, UQUOTA_T_FMT " %d", &quota->used, &quota->limit);
+	if (!*data)
+	    return IMAP_QUOTAROOT_NONEXISTENT;
+	else if (sscanf(data, UQUOTA_T_FMT " %d",
+			&quota->used, &quota->limit) != 2) {
+	    syslog(LOG_ERR, "DBERROR: error fetching quota "
+			    "root=<%s> value=<%s> error=<%s>",
+		   quota->root, data, cyrusdb_strerror(r));
+	    return CYRUSDB_IOERROR;
+	}
 	break;
 
     case CYRUSDB_AGAIN:
